sensor/kubernetes/listener/resources/networkpolicy.go

```diff
@@ -69,7 +69,9 @@ func (h *networkPolicyDispatcher) getSelector(np, oldNp *storage.NetworkPolicy)
 }
 
 func (h *networkPolicyDispatcher) updateDeploymentsFromStore(np *storage.NetworkPolicy, sel selector) {
+	var idsRequireReprocessing []string
 	for _, deploymentWrap := range h.deploymentStore.getMatchingDeployments(np.GetNamespace(), sel) {
-		h.detector.ProcessDeployment(deploymentWrap.GetDeployment(), central.ResourceAction_UPDATE_RESOURCE)
+		idsRequireReprocessing = append(idsRequireReprocessing, deploymentWrap.GetId())
 	}
```

Comment:
[nit] I would initialize `idsRequireReprocessing` so that way we don't have to resize the slice in each iteration.

```go
deployments := h.deploymentStore.getMatchingDeployments(np.GetNamespace(), sel)
idsRequireReprocessing := make([]string, 0, len(deployments))
for _, deploymentWrap := range deployments {
    idsRequireReprocessing = append(idsRequireReprocessing, deploymentWrap.GetId())
}
```